### PR TITLE
[plugin-mobile-app] Add possibility to swipe in LEFT/RIGHT direction

### LINKED ISSUE #1796: Add step to check right and left swipes
Please either **extend** existing step _When I swipe direction to element located `locator` with duration swipeDuration by_ adding right and left swipes or **add** a new step for **right and left swipe**

### PR DESCRIPTION
**Pull request by ikalinin1**
Closes #1796

### CODE REVIEW

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/configuration/MobileApplicationConfiguration.java [hunk 1]
```diff
@@ -25,16 +25,24 @@
...
         this.swipeStabilizationDuration = swipeStabilizationDuration;
         this.swipeLimit = swipeLimit;
-        Validate.isTrue(swipeVerticalXPosition >= 0 && swipeVerticalXPosition <= 100,
-            "The x percentage value must be between 0 and 100, but got: %d", swipeVerticalXPosition);
         this.swipeVerticalXPosition = swipeVerticalXPosition;
+        this.swipeHorizontalYPosition = swipeHorizontalYPosition;
+    }
+
+    @SuppressWarnings("MagicNumber")
+    private void validatePositiom(String name, int value)
```
Comment (uarlouski) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/configuration/MobileApplicationConfiguration.java:42]: validatePositio>n<

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeDirection.java [hunk 2]
```diff
@@ -17,49 +17,81 @@
...
+        {
+            int indent = dimension.getWidth() / HORIZONTAL_INDENT_COEFFICIENT;
+            int y = calculateCoordinate(dimension.getHeight(),
+                    mobileApplicationConfiguration.getSwipeHorizontalYPosition());
+            return new SwipeCoordinates(indent, y, dimension.getWidth() - indent, y);
         }
     };
 
     private static final int VERTICAL_INDENT_COEFFICIENT = 5;
+    private static final int HORIZONTAL_INDENT_COEFFICIENT = 8;
```
Comment (uarlouski) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeDirection.java:72]: why 8? :)

File: vividus-tests/src/main/resources/data/tables/system/mobile_app/locators/ios.table [hunk 3]
```diff
@@ -16,3 +16,6 @@
 |nameInputXpath             |//XCUIElementTypeTextField[@value='Enter your name...']|
 |action                     |COMPARE_AGAINST                                        |
 |scrollViewXpath            |//XCUIElementTypeScrollView                            |
+|carouselViewXpath          |//XCUIElementTypeButton[@name='Carousel']            |
+|firstItemAccessibilityId   |Item 1                                                 |
+|secondItemAccessibilityId  |Item 2                                                 |
```
Comment (uarlouski) [inline vividus-tests/src/main/resources/data/tables/system/mobile_app/locators/ios.table:21]: should we really move identical ids out into the table?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeDirection.java (hunk 2 above)
Comment (ikalinin1) [reply vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeDirection.java:72]: Same logic as for 5

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeDirection.java [hunk 4]
```diff
@@ -17,49 +17,80 @@
...
     public static SwipeCoordinates createCoordinates(int startY, int endY, int width, int xOffsetPercentage)
     {
-        int x = calculateX(width, xOffsetPercentage);
+        int x = calculateCoordinate(width, xOffsetPercentage);
         return new SwipeCoordinates(x, startY, x, endY);
     }
 
     @SuppressWarnings("MagicNumber")
-    private static int calculateX(int width, int percentage)
+    private static int calculateCoordinate(int dimenwion, int percentage)
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeDirection.java:84]:
```suggestion
    private static int calculateCoordinate(int dimension, int percentage)
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**